### PR TITLE
git-annex: Support for installing assistant autostart service.

### PR DESCRIPTION
**Pull request by lhunath**
This adds a plist to ensure `git-annex`'s assistant is started and looks at the `~/.config/git-annex/autostart` file that is created by `git-annex` when you check the setting in the webapp to automatically start the assistant for the repo.

### CODE REVIEW

**Comment by lhunath** [comment]
What's the process for getting this improvement merged?

**Comment by lhunath** [comment]
I think I'll keep bumping this until I get an idea of how to proceed with getting this merged.

File: Formula/git-annex.rb [hunk 1]
```diff
@@ -40,6 +40,30 @@ def install
     bin.install_symlink "git-annex" => "git-annex-shell"
   end
 
+  plist_options :startup => true, :manual => "git annex assistant --autostart"
```
Comment (ilovezfs) [inline Formula/git-annex.rb]: does it really need root privileges? If not the `:startup => true` should go

**Review by ilovezfs — CHANGES_REQUESTED** [review]

File: Formula/git-annex.rb (hunk 1 above)
Comment (lhunath) [reply Formula/git-annex.rb]: I may not fully understand the implications of these options.  I judged off of:
https://www.rubydoc.info/github/Homebrew/brew/Formula.plist_options

The rational being that the service should auto-start on start-up, not wait for the user to manually bring it up each time they boot.

Comment (ilovezfs) [reply Formula/git-annex.rb]: It'll load when the user logs. Note that `httpd` and `php` formulae both use `plist_options :manual`

File: Formula/git-annex.rb [hunk 2]
```diff
@@ -40,6 +40,30 @@ def install
...
+        <false/>
+        <key>ProgramArguments</key>
+        <array>
+          <string>#{opt_bin}/git-annex</string>
+          <string>assistant</string>
+          <string>--autostart</string>
+        </array>
+      </dict>
+    </plist>
+    EOS
```
Comment (ilovezfs) [inline Formula/git-annex.rb]: it's going to want this shifted two spaces to the left because it's mean

**Review by ilovezfs — APPROVED** [review]

**Comment by lhunath** [comment]
Thanks for your guidance!

**Comment by ilovezfs** [comment]
Shipped! Thanks for your patience @lhunath and another contribution.

**Pull request merged**